controllers/machinehealthcheck_controller.go

```diff
@@ -452,3 +530,18 @@ func machineNames(machines []*clusterv1.Machine) []string {
 	}
 	return result
 }
+
+// externalRemediationRequestExists checks if the External Remediation Request is created
+// for the machine.
+func (r *MachineHealthCheckReconciler) externalRemediationRequestExists(ctx context.Context, m *clusterv1.MachineHealthCheck, machineName string) bool {
```

Comment:
Can we:
- Change this to `getExternalRemediationRequest` and have it return `(*unstructured.Unstructured, error)`
- Add another method for `externalRemediationRequestExists` that calls `getExternalRemediationRequest` and returns a `bool`
- Use `getExternalRemediationRequest` above in the loop where you're deleting remediation requests for healthy machines

?

Reply:
@ncdc please review the changes.